Apache Flume Initial Package 1.9.0

polynomial opened the pull request:
###### Motivation for this change

We use Apache Flume in production and are migrating everything to NixOS. I will write a module next, but this gets me a working Apache Flume jar, which is a great first step!

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

polynomial committed:
Apache Flume Initial Package 1.9.0

volth commented on pkgs/servers/apache-flume/default.nix:
```diff
@@ -0,0 +1,48 @@
+    inherit sha256;
+  };
+
+  buildInputs = [ hadoop jre makeWrapper bash gnused coreutils ];
```
```bash``` ```gnused``` ```coreutils``` are already included in every bulid by default

polynomial replied:
Fixed, thanks! Old habit's die hard (:

polynomial committed:
Apache Flume Initial Module

polynomial commented:
What are the steps to get this merged? I have addressed the comment and am using this locally, would love to see it be part of the tree. I am currently working on a module and will submit a PR for that shortly.

JohnAZoidberg commented on pkgs/top-level/all-packages.nix:
```diff
@@ -8580,6 +8580,9 @@ in
   apacheKafka_2_2 = callPackage ../servers/apache-kafka { majorVersion = "2.2"; };
 
+  apacheFlume = apacheFlume_1_9_0;
+  apacheFlume_1_9_0 = callPackage ../servers/apache-flume { majorVersion = "1.9.0"; };
```
Do you think there will be reasons to keep multiple versions of flume?
It does not seem like there are multiple maintained at the same time.
If that's not the case, it'd be better not to have to maintain multiple versions for us.

JohnAZoidberg commented on pkgs/servers/apache-flume/default.nix:
```diff
@@ -0,0 +1,45 @@
+    homepage = http://flume.apache.org;
+    description = "Flume is a distributed, reliable, and available service for efficiently collecting, aggregating, and moving large amounts of log data.";
+    license = licenses.asl20;
+    maintainers = [ ];
```
Would you be willing to look after this package, i.e. be maintainer?

JohnAZoidberg commented on pkgs/servers/apache-flume/default.nix:
```diff
@@ -0,0 +1,45 @@
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = http://flume.apache.org;
```
```suggestion
    homepage = "https://flume.apache.org";
```
We'll deprecate unquoted URL syntax, once RFC 45 is approved. Also their site supports https.

JohnAZoidberg commented on pkgs/servers/apache-flume/default.nix:
```diff
@@ -0,0 +1,45 @@
+    wrapProgram $out/bin/flume-ng \
+      --set FLUME_HOME "$out" \
+      --set JAVA_HOME "${jre}"
+    chmod +x $out/bin\/*
```
I'm pretty sure `wrapProgram` already marks its output as executable. Please try to remove it.

JohnAZoidberg commented on pkgs/servers/apache-flume/default.nix:
```diff
@@ -0,0 +1,45 @@
+
+stdenv.mkDerivation rec {
+  version = "${flumeVersion}";
+  name = "apache-flume-${version}";
```
```suggestion
  pname = "apache-flume";
```

`pname` appends the version name on its own.

JohnAZoidberg commented on pkgs/servers/apache-flume/default.nix:
```diff
@@ -0,0 +1,45 @@
+with versionMap.${majorVersion};
+
+stdenv.mkDerivation rec {
+  version = "${flumeVersion}";
```
```suggestion
  version = flumeVersion;
```
No string quoting necessary here :)

JohnAZoidberg commented on pkgs/servers/apache-flume/default.nix:
```diff
@@ -0,0 +1,45 @@
+
+  meta = with stdenv.lib; {
+    homepage = http://flume.apache.org;
+    description = "Flume is a distributed, reliable, and available service for efficiently collecting, aggregating, and moving large amounts of log data.";
```
Please follow https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

SuperSandro2000 commented:
Closing due to being stale.
